tests/bluetooth/bsim_bt/bsim_test_l2cap/src/main_l2cap_ecred.c

```diff
@@ -0,0 +1,573 @@
+/* main_l2cap_ecred.c - Application main entry point */
+
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common.h"
+
+#define LOG_MODULE_NAME main_l2cap_ecred
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_DBG);
+
+extern enum bst_result_t bst_result;
+
+static struct bt_conn *default_conn;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+};
+
+#define DATA_MTU 2000
+#define DATA_MPS 65
+#define DATA_BUF_SIZE BT_L2CAP_SDU_BUF_SIZE(DATA_MTU)
+#define L2CAP_CHANNELS 2
+#define SERVERS 1
+#define SDU_SEND_COUNT 200
+#define ECRED_CHAN_MAX 5
+
+NET_BUF_POOL_FIXED_DEFINE(rx_data_pool, L2CAP_CHANNELS, BT_L2CAP_BUF_SIZE(DATA_BUF_SIZE), 8, NULL);
+NET_BUF_POOL_FIXED_DEFINE(tx_data_pool, L2CAP_CHANNELS + 1, BT_L2CAP_BUF_SIZE(DATA_MTU), 8, NULL);
+
+static struct bt_l2cap_server servers[SERVERS];
+void send_sdu_chan_worker(struct k_work *item);
+
+static struct channel {
+	uint8_t chan_id; /* Internal number that identifies L2CAP channel. */
+	struct bt_l2cap_le_chan le;
+	bool in_use;
+	int sdus_received;
+	int bytes_to_send;
+	int iteration;
+	struct net_buf *buf;
+	struct k_work work;
+	uint8_t payload[DATA_MTU];
+} channels[L2CAP_CHANNELS];
+
+CREATE_FLAG(is_connected);
+CREATE_FLAG(unsequenced_data);
+
+#define MY_STACK_SIZE 512
+#define MY_PRIORITY 5
+
+static K_THREAD_STACK_ARRAY_DEFINE(my_stack_area, L2CAP_CHANNELS, MY_STACK_SIZE);
+static struct k_work_q my_work_q[L2CAP_CHANNELS];
```

Comment:
Is there a 1:1 between `my_work_q[x]` and `channels[x]`? If so, this can probably be moved into the struct as well.

Reply:
I tried to put it into the channel's struct but it is not working :(

Reply:
It should work, and if it doesn't due to some issue, then I think the issue lies somewhere else, and we should be able to find and fix that.

Reply:
I am running out of ideas why it is not working in that way, could we just accept it ?

Reply:
I'm not happy about accepting it, as it could point towards another issue in this case; possibly one that is timing related as that I is the only theory I have given the simplicity of the change.